src/popup/App.vue

```diff
@@ -591,54 +591,72 @@ export default {
       let that = this;
       that.loading = true;
 
-      let res = await fetch(
-        that.apiUrl + "/api/v2/alias/custom/new?hostname=" + that.hostName,
-        {
-          method: "POST",
-          body: JSON.stringify({
-            alias_prefix: that.aliasPrefix,
-            signed_suffix: that.signedSuffix,
-          }),
-          headers: {
-            "Content-Type": "application/json",
-            Authentication: this.apiKey,
-          },
-        }
-      );
-
-      let json = await res.json();
-      that.loading = false;
-      if (res.status == 201) {
-        that.newAlias = json.alias;
-        that.recomputeShowVoteScreen();
-      } else {
-        that.showError(json.error);
-      }
+      axios
+        .post(
+          that.apiUrl + "/api/v2/alias/custom/new?hostname=" + that.hostName,
+          { alias_prefix: that.aliasPrefix, signed_suffix: that.signedSuffix },
+          {
+            headers: { Authentication: this.apiKey },
+          }
+        )
+        .then((res) => {
+          if (res.status == 201) {
+            that.newAlias = res.data.alias;
+            that.recomputeShowVoteScreen();
+          } else {
+            that.showError(res.data.error);
+          }
+        })
+        .catch((err) => {
+          // rate limit reached
+          if (err.request.status == 429) {
+            that.showError(
```

Comment:
Just a small question: since we are using arrow function, we can use "this" to refer to original context. So, should we change "that" to "this" (or maybe we will do with refactoring?)

Reply:
Good point! `that` is not needed here indeed, I just updated the PR to replace `that` by `this`